Backend/Remora.Discord.Caching/Extensions/CacheBuilder.cs

```diff
@@ -0,0 +1,73 @@
+//
+//  CacheBuilder.cs
+//
+//  Author:
+//       Jarl Gullberg <jarl.gullberg@gmail.com>
+//
+//  Copyright (c) 2017 Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using Microsoft.Extensions.DependencyInjection;
+using Remora.Discord.Caching.Clients;
+
+namespace Remora.Discord.Caching.Extensions
+{
+    /// <summary>
+    /// Utility class used by <see cref="ServiceCollectionExtensions.AddDiscordCaching"/> to configure the cache.
+    /// </summary>
+    public class CacheBuilder
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CacheBuilder"/> class.
+        /// </summary>
+        /// <param name="services">Services of the application.</param>
+        public CacheBuilder(IServiceCollection services)
+        {
+            Services = services;
+        }
+
+        /// <summary>
+        /// Gets the application services.
+        /// </summary>
+        public IServiceCollection Services { get; }
+
+        /// <summary>
+        /// Replaces the default memory cache client with the given type.
+        /// </summary>
+        /// <remarks>
+        /// The type will be registered as a singleton in the service provider.
+        /// </remarks>
+        /// <typeparam name="TClient">New cache client to use.</typeparam>
+        /// <returns>Same instance of the builder.</returns>
+        public CacheBuilder UseClient<TClient>() where TClient : class, ICacheClient
+        {
+            Services.AddSingleton<ICacheClient, TClient>();
+            return this;
+        }
+
+        /// <summary>
+        /// Replaces the default memory cache client with the distributed cache client.
```

Comment:
This method doesn't actually replace anything; it just adds another more prioritized service. Either correct the comment or the implementation.

Reply:
Changed, is this okay?

Reply:
Yep, looks fine :)